.semgrep/rules/sol-rules.yaml

```diff
@@ -195,3 +195,45 @@ rules:
     paths:
       exclude:
         - packages/contracts-bedrock/test
+
+  - id: sol-safety-use-disable-initializer
+    languages: [solidity]
+    severity: ERROR
+    message: Proxied contracts (excluding predeploys) must disable initializers in constructor
+    patterns:
+      - pattern-regex: "///\\s*@custom:proxied\\s+true(?P<CONTRACT>[\\s\\S]*)"
+      - pattern-not-regex: "///\\s*@custom:predeploy.*(?P<REST>[\\s\\S]*)"
+      - focus-metavariable: $CONTRACT
+      - pattern: |
+          constructor(...) {
+            ...
+          }
+      - pattern-not: |
+          constructor(...) {
+            ...
+            _disableInitializers();
+            ...
+          }
+    paths:
+      exclude:
+        - packages/contracts-bedrock/src/L1/SystemConfigInterop.sol
+        - packages/contracts-bedrock/src/L1/OptimismPortalInterop.sol
+
+  - id: sol-safety-single-proper-initializer
```

Comment:
Sorry one more change here, let's make this test be called `sol-safety-proper-initializer` and then we need a third check for `sol-safety-single-initializer`. First let's change the name of this check.

Reply:
What's the distinction between the two?

Reply:
Basically one checks that you have the correct initializer style and the other checks that you have only one initializer. It would be too annoying to do them as one check.